main.tf

```diff
@@ -1,21 +1,40 @@
-data "aws_caller_identity" "default" {}
+locals {
+  e = module.this.enabled # Shorthand for quick reference
+
+  sns_topic_policy_enabled = local.e && !var.add_sns_policy && var.sns_topic_arn != ""
+  sns_topic_arn            = var.add_sns_policy && var.sns_topic_arn != "" ? var.sns_topic_arn : join("", aws_sns_topic.default.*.arn)
+  endpoints                = distinct(compact(concat([local.sns_topic_arn], var.additional_endpoint_arns)))
+}
+
+data "aws_caller_identity" "default" {
+  count = module.this.enabled ? 1 : 0
```

Comment:
If we're going to define the local, let's change all usages of `module.this.enabled` to use `local.enabled` instead.

Reply:
Good suggestion, I also added the usage of `count = local.enabled ? 1 : 0` in the resources which were missing it.